zfs/zfs.go

```diff
@@ -779,17 +807,25 @@ func ZFSSend(ctx context.Context, sendArgs ZFSSendArgsValidated) (*SendStream, e
 	args = append(args, "send")
 
 	// pre-validation of sendArgs for plain ErrEncryptedSendNotSupported error
+	// we tie BackupProperties (send -b) and SendRaw (-w, same as with Encrypted) to this
+	// since these were released together.
 	// TODO go1.13: push this down to sendArgs.Validate
-	if encryptedSendValid := sendArgs.Encrypted.Validate(); encryptedSendValid == nil && sendArgs.Encrypted.B {
-		supported, err := EncryptionCLISupported(ctx)
+
+	if sendArgs.Encrypted.B || sendArgs.Raw || sendArgs.BackupProperties {
+		encryptionSupported, err := EncryptionCLISupported(ctx)
 		if err != nil {
 			return nil, errors.Wrap(err, "cannot determine CLI native encryption support")
 		}
-		if !supported {
+
+		if !encryptionSupported {
 			return nil, ErrEncryptedSendNotSupported
 		}
 	}
 
+	if _, err := sendArgs.Validate(ctx); err != nil {
```

Comment:
This doesn't make sense sinse sendArgs are already `ZFSSendArgsValidated`.

Was this supposed to be the `sendArgs.Encrypted.Validate()` call? (see left side of this diff)

Reply:
Maybe, I kinda questioned myself about this chunk while rebasing.
Question though: Why do we revalidate Encrypted (in the original code) here? It's already validated in `ZFSSendArgsUnvalidated.Validate()`.
The comment mentions go1.13? Is this an oversight?